extensions/amp-riddle-quiz/0.1/amp-riddle-quiz.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getData, listen} from './../../../src/event-helper';
+import {isObject, isFiniteNumber} from '../../../src/types';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {user} from '../../../src/log';
+
+export class AmpRiddleQuiz extends AMP.BaseElement {
+
+    /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?number} */
+    this.itemHeight_ = 300; //default
+
+    /** @private {?number} */
+    this.riddleId_ = null;
+
+    /** @private {!Element} */
+    this.container_ = this.win.document.createElement('div');
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+  }
+
+  handleMessage_(event) {
+    if (event.origin != 'https://www.riddle.com' ||
```

Comment:
please also add : `if( !this.iframe_ || event.origin ....`

This is because iframe can get nullified in `unlayoutcallback` but messages might be quieted and received after nullification in rare edge cases.